tools/cereal-scaffold/main.go

```diff
@@ -224,147 +449,96 @@ func (t *SimpleTask) Run(ctx context.Context, task cereal.Task) (interface{}, er
 	return params.ID, nil
 }
 
+type SimpleWorkflow struct{}
+
 type SimpleWorkflowParams struct {
-	NumTasks int
+	NumTasks  int
+	SlowTasks bool
 }
 
-type SimpleWorkflow struct{}
+type SimpleWorkflowPayload struct {
+	Counter   int
+	StartTime time.Time
+}
 
 func (p *SimpleWorkflow) OnStart(w cereal.WorkflowInstance,
 	ev cereal.StartEvent) cereal.Decision {
 
 	logrus.Info("SimpleWorkflow got OnStart")
-
 	params := SimpleWorkflowParams{}
 	err := w.GetParameters(&params)
 	if err != nil {
-		panic(err)
+		return w.Fail(errors.Wrap(err, "could not decode workflow parameters"))
 	}
 	if params.NumTasks == 0 {
 		logrus.Error("Got no tasks to do")
 		return w.Complete()
 	}
 
 	for i := 0; i < params.NumTasks; i++ {
-		w.EnqueueTask("simple task", &SimpleTaskParams{ID: fmt.Sprintf("asdf: %d", i), Sleepy: i * 2})
+		err = w.EnqueueTask("simple task", &SimpleTaskParams{ID: fmt.Sprintf("asdf: %d", i), Sleepy: i * 2, SlowTasks: params.SlowTasks})
+		if err != nil {
+			return w.Fail(errors.Wrap(err, "could not enqueue task"))
+		}
 	}
 
-	initialVal := 0
-	return w.Continue(&initialVal)
+	return w.Continue(&SimpleWorkflowPayload{
+		Counter:   0,
+		StartTime: time.Now(),
```

Comment:
Putting this here means we don't measure the OnStart time, but for what we typically care about when actually looking at this time, that doesn't matter.